src/unit_selection.rs

```diff
@@ -0,0 +1,143 @@
+    pub fn current_to_last(&mut self)
+    {
+        self.last_selected = self.currently_selected;
+    }
+}
+
+pub struct SelectionColor(pub Color);
+
+impl SelectionColor
+{
+    const SELECTED_COLOR: Color = Color::rgb(0.3, 0.8, 0.5);
+}
+
+impl Default for SelectionColor
+{
+    fn default() -> Self {
+        SelectionColor(Self::SELECTED_COLOR)
+    }
+}
+
+
+fn selection_update(
+    pick_state: Res<PickState>,
+    mut selection_state: ResMut<SelectionState>,
+    mut selection_events: ResMut<Events<SelectionEvent>>,
+    mouse_button_inputs: Res<Input<MouseButton>>,
+    query: Query<&Selectable>
+)
+{
+    if mouse_button_inputs.just_pressed(MouseButton::Left)
+    {
+        if let Some(pick_depth) = pick_state.top()
+        {
+            selection_state.swap_and_clear();
+            
+            if let Ok(_) = query.get::<Selectable>(pick_depth.entity())
+            {
+                selection_state.currently_selected = Some(pick_depth.entity());
+                selection_events.send(SelectionEvent {
+                    selected: pick_depth.entity()
+                })
+            }
+        }
+    }
+}
+
+#[derive(Default)]
+struct LastSelectedColor(pub Color);
+
+fn highlight_selected(
+    mut local_last_selected_color: Local<LastSelectedColor>,
+    mut selection_reader: ResMut<SelectionEventReader>,
+    selection_events: Res<Events<SelectionEvent>>,
+    selection_state: Res<SelectionState>,
+    selection_color: Res<SelectionColor>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    query: Query<&mut Handle<StandardMaterial>>
+)
+{
+    // If a selection event has occured, then do stuff to highlight the selected mesh
+    if let Some(event) = selection_reader.reader.latest(&selection_events)
+    {
+        println!("Found selection event: {:?}", event);
+
+        // If there's something that was previously selected, then we need to set it's material
+        // back to it's original material.
+        match selection_state.last_selected
+        {
+            Some(entity) => {
+                match query.get_mut::<Handle<StandardMaterial>>(entity)
+                {
+                    Ok(handle) => {
+                        materials.get_mut(&handle).unwrap().albedo = local_last_selected_color.0;
+                    },
+                    Err(error) => println!("Highlight selected entity query error: {:?}", error)
+                }
+            },
+            None => {}
+        }
+        
+        // Now set the selected entity's material to the selected material.
+        match query.get_mut::<Handle<StandardMaterial>>(event.selected)
+        {
+            Ok(handle) => {
+                let color = &mut materials.get_mut(&handle).unwrap().albedo;
+
+                local_last_selected_color.0 = *color;
+                *color = selection_color.0;
+            },
+            Err(error) => println!("Highlight selected entity query error: {:?}", error)
```

Comment:
```suggestion
            Err(error) => eprintln!("Highlight selected entity query error: {:?}", error)
```
This would be an error. In fact, I'd argue for a `panic` here as it really suggests something is wrong, but at the least `eprintln` is appropriate.